src/hwProvider.ts

```diff
@@ -43,23 +44,69 @@ export class HWProvider {
     }
 
     async getTransport(): Promise<Transport> {
-        let webUSBSupported = await TransportWebUSB.isSupported();
-        webUSBSupported =
-            webUSBSupported &&
-            platform.name !== "Opera";
+        const isLedgerSupported = await this.isLedgerTransportSupported();
+
+        if (!isLedgerSupported) {
+            throw Error('Ledger is not supported');
+        }
+
+        let webBLESupported = await this.isBLESupported();
+
+        if (webBLESupported) {
+            return await TransportWebBLE.create();
+        }
+
+        let webUSBSupported = await this.isWebUSBSupported();
 
         if (webUSBSupported) {
             return await TransportWebUSB.create();
         }
 
-        let webHIDSupported = await TransportWebHID.isSupported();
+        let webHIDSupported = await this.isWebUSBSupported();
+
         if (webHIDSupported) {
-            return await TransportWebHID.open("");
+            return await TransportWebHID.create();
         }
 
+        // Deprecated: Use WebHID instead
+        // https://developers.ledger.com/docs/transport/choose-the-transport/
         return await TransportU2f.create();
     }
 
+    async isLedgerTransportSupported(): Promise<boolean> {
```

Comment:
Not that the separate `isStuffSupported` are not fast, but still, it would be more optimal to do
```js
return await this.isBLESupported() || await this.[...] ...